src/mqt/qecc/circuit_synthesis/elimination.py

```
@@ -0,0 +1,719 @@
+        self.operations.append(op)
+        # find maximum depth of involved qubits
+        involved_qubits = op.qubits()
+        earliest_start = 0
+        for q in involved_qubits:
+            if q in self._qubit_depths:
+                earliest_start = max(earliest_start, self._qubit_depths[q] + 1)
+        for q in involved_qubits:
+            self._qubit_depths[q] = earliest_start
+        self._depth = max(self._depth, earliest_start + 1)
+
+    def apply(self, tableau: BinaryMatrix, inplace: bool = False) -> BinaryMatrix:
```

Ruff (0.15.12)

[warning] Boolean-typed positional argument in function definition

(FBT001)